### ISSUE TITLE
Add an option to apply SavedModel via command line 

### ISSUE THREAD

**Issue by goswamig** [tensorflow/tensorflow#23207]
**System information**
- TensorFlow version (you are using): TF-1.11
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
By default TensorFlow has an option to store checkpoint by adding '--train_dir'. However this is good when we want to continue training, but if we want to perform inference using TensorFlowServing it becomes a challenge.  Is it possible to add some flag which will store the variables, graph, in/output signature by just providing an option similar to '--train_dir'?

**Will this change the current api? How?**
Yes, but nothing will break 

for e.g. **Old API** will store only checkpoints
tf_cnn_benchmarks.py --model resnet50 --num_epochs=2 --batch_size 64 --data_name=imagenet  --data_dir=/mnt/data **--train_dir=/mnt/checkpoint**  --variable_update horovod --horovod_device gpu --weight_decay=1e-4 --use_fp16

**New API** should store using SaveModel
tf_cnn_benchmarks.py --model resnet50 --num_epochs=2 --batch_size 64 --data_name=imagenet  --data_dir=/mnt/data **--train_dir_save_model=/mnt/checkpoint**  --variable_update horovod --horovod_device gpu --weight_decay=1e-4 --use_fp16

**Who will benefit with this feature?**
By having such option, deploying the model to production will be super fast, even for the people who doesn't know anything about training code. Currently we have to either change the training code to use SaveModel or figure out a way to export graph along with variables. 

**Any Other info.**
N/A

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by gautamkmr** [comment]
I do understand that this will be hard to implement but at least having such option in existing example specially the benchmark example will be really useful.

Imagine a scenario where an user not a TF expert (for e.g. me), I can just run your resnet50 from tf_cnn_benchmark code on imagenet dataset. Use the output of training for inference on any image detection system that I want to build. I can't do that today that easily because training is producing the output in the form of checkpoint which can't be consumed by TF serving.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @karmel: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by karmel** [comment]
Given that this is a feature request specific to tf_cnn_benchmarks, reassigning to @reedwm

**Comment by reedwm** [comment]
tf_cnn_benchmarks is not designed for production use, which is why it doesn't supported SavedModel. That said, we will accept PRs to tf_cnn_benchmarks implementing SavedModel support if they are relatively simple, easy to maintain, and have tests.

@karmel, is the official resnet50 model designed for production? Perhaps it should support SavedModel

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by frreiss** [comment]
This issue should be under the [TensorFlow benchmarks](https://github.com/tensorflow/benchmarks) project, not the main TensorFlow core project.

**Comment by Sudeepam97** [comment]
If that is so, then maybe someone with the appropriate permissions should transfer this issue to the TensorFlow benchmarks project. [This](https://help.github.com/articles/transferring-an-issue-to-another-repository/) page shows how it's done.
